tracer/src/Datadog.Trace.Tools.Runner/Utils.cs

```diff
@@ -280,5 +282,59 @@ public static string GetEnvironmentVariable(string key, string defaultValue = nu
 
             return defaultValue;
         }
+
+        public static bool CheckAgentConnection(string agentUrl)
+        {
+            var env = new Dictionary<string, string>();
+            if (!string.IsNullOrWhiteSpace(agentUrl))
+            {
+                env["DD_TRACE_AGENT_URL"] = agentUrl;
+            }
+
+            var tracerSettings = new Configuration.TracerSettings(new DictionaryConfigurationSource(env));
+
+            try
+            {
+                var agentWriter = new CIAgentWriter(tracerSettings);
+
+                if (agentWriter.Ping().GetAwaiter().GetResult())
+                {
+                    Console.WriteLine($"Connection with the Datadog Agent at {tracerSettings.AgentUri} was successful.");
+                }
+                else
+                {
+                    Console.WriteLine($"Error connecting to the Datadog Agent at {tracerSettings.AgentUri}.");
+                    return false;
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error connecting to the Datadog Agent at {tracerSettings.AgentUri}.");
+                Console.WriteLine(ex);
+                return false;
+            }
+
+            return true;
+        }
+
+        private class DictionaryConfigurationSource : Configuration.StringConfigurationSource
```

Comment:
Any reason for not using the built-in `NameValueConfigurationSource`?

Reply:
You're right, this is not required. Done in 81f0c7e